code/game/objects/mail.dm

```diff
@@ -0,0 +1,233 @@
+	pickup_sound =  'sound/items/handling/paper_pickup.ogg'
+	var/list/possible_contents = list()
+	var/list/job_list = list()
+	var/mob/living/recipient
+	var/has_been_scanned = FALSE
+
+/obj/item/envelope/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is licking a sharp corner of the envelope. It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	playsound(loc, 'sound/effects/-adminhelp.ogg', 50, TRUE, -1)
+	return BRUTELOSS
+
+/obj/item/envelope/attack_self(mob/user)
+	if(user.mind.current != recipient.mind.current)
+		to_chat(user, "<span class='warning'>You don't want to open up another person's mail, that's an invasion of their privacy!</span>")
+		return
+	if(do_after(user, 1 SECONDS, target = user) && !QDELETED(src))
+		to_chat(user, "<span class='notice'>You begin to open the envelope.</span>")
+		playsound(loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+		user.unEquip(src)
+		for(var/obj/item/I in contents)
+			user.put_in_hands(I)
+		qdel(src)
+
+/obj/item/envelope/Initialize(mapload)
+	. = ..()
+	var/item = pick(possible_contents)
+	new item(src)
+	new /obj/item/stack/spacecash(src, rand(1, 50) * 5)
+	var/list/mind_copy = SSticker.minds.Copy()
+	shuffle(mind_copy)
+	for(var/datum/mind/mail_attracted_people in mind_copy)
+		var/turf/T = get_turf(mail_attracted_people)
+		if(mail_attracted_people.offstation_role || !ishuman(mail_attracted_people) || T.z == 1)
+			continue
+		if(mail_attracted_people.assigned_role in job_list)
+			recipient = mail_attracted_people.current
+			name = "letter to [recipient.real_name]"
+			return
+	if(!admin_spawned)
+		log_debug("Error: failed to find a new name to assign to [src]!")
+		qdel(src)
+
+/obj/item/envelope/security
+	icon_state = "mail_sec"
+	possible_contents = list(/obj/item/reagent_containers/food/snacks/donut/sprinkles,
+							/obj/item/megaphone,
+							/obj/item/poster/random_official,
+							/obj/item/restraints/handcuffs/pinkcuffs,
+							/obj/item/restraints/legcuffs/bola/energy,
+							/obj/item/reagent_containers/food/drinks/coffee,
+							/obj/item/stock_parts/cell/super,
+							/obj/item/grenade/barrier/dropwall,
+							/obj/item/toy/figure/crew/detective,
+							/obj/item/toy/figure/crew/hos,
+							/obj/item/toy/figure/crew/secofficer)
+	job_list = list("Head of Security", "Security Officer", "Detective", "Warden")
+
+/obj/item/envelope/science
+	icon_state = "mail_sci"
+	possible_contents = list(/obj/item/analyzer,
+	/obj/item/assembly/signaler,
+	/obj/item/slime_extract/grey,
+	/obj/item/clothing/mask/gas,
+	/obj/item/reagent_containers/spray/cleaner,
+	/obj/item/clothing/glasses/regular,
+	/obj/item/taperecorder,
+	/obj/item/paicard,
+	/obj/item/toy/figure/crew/borg,
+	/obj/item/toy/figure/crew/geneticist,
+	/obj/item/toy/figure/crew/rd,
+	/obj/item/toy/figure/crew/roboticist,
+	/obj/item/toy/figure/crew/scientist)
+	job_list = list("Research Director", "Roboticist", "Geneticist",  "Scientist")
```

Comment:
if https://github.com/ParadiseSS13/Paradise/pull/22328 goes in before this PR, I'd definitely recommend looking at reworking the job list to use this instead or in addition

Reply:
Will it? Currently it isn't in, and I can't really make changes based off of a PR that may or may not be merged in its current state.

Reply:
I'm not saying you need to make changes now. I'm just suggesting that if it *does* go in, you might be able to take advantage of its benefits, and I wanted to make sure you were aware of it for that reason

Reply:
I'll keep that in mind thank you.